src/foam/nanos/script/jShell/InstructionPresentation.java

```diff
@@ -24,17 +24,30 @@ public InstructionPresentation(JShell jShell) {
   }
 
   public List<String> parseToInstruction(List<String> scripts) {
-    int i = 0;
+    int    i           = 0;
     String codeToParse = "";
     while ( i < scripts.size() ) {
-      codeToParse += scripts.get(i);
-      SourceCodeAnalysis.CompletionInfo info = jShell.sourceCodeAnalysis().analyzeCompletion(codeToParse);
-      if ( info.completeness().isComplete() ) {
+      String instraction = checkEmptyInstraction(scripts.get(i));
+      codeToParse += instraction;
+      SourceCodeAnalysis.CompletionInfo info = jShell.sourceCodeAnalysis()
+        .analyzeCompletion(codeToParse);
+      if ( info.completeness()
+        .isComplete() ) {
         listInstruction.add(codeToParse);
         codeToParse = "";
       }
       i++;
     }
     return listInstruction;
   }
+
+  private String checkEmptyInstraction(String instraction) {
+    if ( instraction.startsWith("//") ) instraction = "";
+
+    if ( instraction.contains("//") ) {
```

Comment:
this ignores a case where the double slashes could be within a string. in which case, it would cause issues parsing the script. this line of code itself illustrates the problem:
```
if ( instraction.contains("//") ) {
````
would end up being returned as
```
if ( instraction.contains("
```